src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -2,911 +2,591 @@
         {
-            int charCount = jpDecoder.GetChars(japaneseEncodedBuffer, byteIndex, 1, buffer, charIndex);
-            if (charCount > 0)
-            {
-                charIndex += charCount;
-            }
-            byteIndex++;
+            return new KeyValuePair<int, string>(codePage, webName);
         }
 
-        int[] expectedResult = new int[] { 0xA, 0x30CA, 0x30CA, 0x30CA, 0x30CA, 0x1B, 0x1, 0x2, 0x3, 0x4, 0x30CA, 0xFF65,
-                                                 0xFF8A, 0x41, 0x42, 0xFF65, 0x43};
-        Assert.True(charIndex == expectedResult.Length);
-        for (int i = 0; i < charIndex; i++)
-            Assert.True(expectedResult[i] == (int)buffer[i]);
-    }
-
-    [Fact]
-    public static void TestDecodingGB18030()
-    {
-        EnsureInitialization();
-
-        string codepageName = "GB18030"; // 54936 codepage 
-
-        byte[] chineseBuffer = new byte[]
-            {
-                0x41, 0x42, 0x43,
-                0x81, 0x40,
-                0x82, 0x80,
-                0x81, 0x30, 0x82, 0x31,
-                0x81, 0x20
-            };
-
-        Decoder chineseDecoder = Encoding.GetEncoding(codepageName).GetDecoder();
-        const int BUFFER_SIZE = 100;
-        char[] buffer = new char[BUFFER_SIZE];
-
-        int byteIndex = 0;
-        int charIndex = 0;
-
-        while (byteIndex < chineseBuffer.Length)
+        [Fact]
+        [ActiveIssue(846, PlatformID.AnyUnix)]
+        public static void TestDefaultEncodings()
         {
-            int charCount = chineseDecoder.GetChars(chineseBuffer, byteIndex, 1, buffer, charIndex);
-            if (charCount > 0)
+            // Check which encodings are available by default.
+            foreach (var mapping in CrossplatformDefaultEncodings())
             {
-                charIndex += charCount;
+                Encoding encoding = Encoding.GetEncoding(mapping.Key);
+                Assert.NotNull(encoding);
+                Assert.Equal(mapping.Value, encoding.WebName);
             }
-            byteIndex++;
-        }
-
-        int[] expectedResult = new int[] { 0x41, 0x42, 0x43, 0x4E02, 0x500B, 0x8B, 0x3F, 0x20 };
-        Assert.True(charIndex == expectedResult.Length);
-        for (int i = 0; i < charIndex; i++)
-            Assert.True(expectedResult[i] == (int)buffer[i]);
-    }
-
-    [Fact]
-    public static void TestDecodingDBCS()
-    {
-        EnsureInitialization();
+            // Currently the class EncodingInfo isn't present in corefx, but this method should check for the exhaustive list
+            // When it is, comment out this line and remove the previous foreach/assert.
+            // Assert.Equal(CrossplatformDefaultEncodings, Encoding.GetEncodings().OrderBy(i => i.CodePage).Select(i => Map(i.CodePage, i.WebName)));
 
-        string codepageName = "shift_jis"; // 932 codepage 
+            // UTF-8 potentially isn't the default everywhere if a user changes a setting, but we can hope!
```

Comment:
You cannot assume UTF8 will be the default encoding. this is can fail depend on the user settings as the comment say. I would like to have the test more reliable